packages/types/src/primitive/Extrinsic/Extrinsic.ts

```diff
@@ -169,18 +165,53 @@ export default class Extrinsic extends Base<ExtrinsicV1> implements IExtrinsic,
     return this.raw.method;
   }
 
+  /**
+   * @description The nonce for this extrinsic
+   */
+  public get nonce (): Nonce {
+    return this.raw.signature.nonce;
+  }
+
   /**
    * @description The [[ExtrinsicSignature]]
    */
-  public get signature (): ExtrinsicSignature {
-    return this.raw.signature;
+  public get signature (): IHash {
+    return this.raw.signature.signature;
+  }
+
+  /**
+   * @description The [[Address]] that signed
+   */
+  public get signer (): Address {
+    return this.raw.signature.signer;
+  }
+
+  /**
+   * @description Forwards compat
+   */
+  public get tip (): Balance {
+    return this.raw.signature.tip;
+  }
+
+  /**
+   * @description Returns the encoded version flag
+  */
+  public get version (): number {
+    return this.versionFormat | (this.isSigned ? BIT_SIGNED : BIT_UNSIGNED);
+  }
+
+  /**
+   * @description Returns the raw version (not flagged)
+  */
+  public get versionFormat (): number {
+    return this.raw.version;
   }
 
   /**
    * @description Add an [[ExtrinsicSignature]] to the extrinsic (already generated)
    */
-  public addSignature (signer: Address | Uint8Array | string, signature: Uint8Array | string, nonce: AnyNumber, era: Uint8Array | IExtrinsicEra): Extrinsic {
-    this.raw.addSignature(signer, signature, nonce, era);
+  public addSignature (signer: Address | Uint8Array | string, signature: Uint8Array | string, payload: Uint8Array | string): Extrinsic {
```

Comment:
Incompatible change. Signature now takes the actual encoded payload, this makes it much easier to support multiple versions all around